pkg/config/aws-ebs/aws-ebs.go

```diff
@@ -0,0 +1,399 @@
+package aws_ebs
+
+import (
+	"fmt"
+	"strings"
+
+	opv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/aws-ebs-csi-driver-operator/pkg/clients"
+	"github.com/openshift/aws-ebs-csi-driver-operator/pkg/config/common"
+	"github.com/openshift/aws-ebs-csi-driver-operator/pkg/generator"
+	"github.com/openshift/aws-ebs-csi-driver-operator/pkg/operator/config"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/csi/csidrivercontrollerservicecontroller"
+	"github.com/openshift/library-go/pkg/operator/csi/csidrivernodeservicecontroller"
+	"github.com/openshift/library-go/pkg/operator/csi/csistorageclasscontroller"
+	dc "github.com/openshift/library-go/pkg/operator/deploymentcontroller"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	cloudCredSecretName   = "ebs-cloud-credentials"
+	metricsCertSecretName = "aws-ebs-csi-driver-controller-metrics-serving-cert"
+	infrastructureName    = "cluster"
+	cloudConfigNamespace  = "openshift-config-managed"
+	cloudConfigName       = "kube-cloud-config"
+	caBundleKey           = "ca-bundle.pem"
+	trustedCAConfigMap    = "aws-ebs-csi-driver-trusted-ca-bundle"
+	kmsKeyID              = "kmsKeyId"
+)
+
+func GetAWSEBSGeneratorConfig() *generator.CSIDriverGeneratorConfig {
+	return &generator.CSIDriverGeneratorConfig{
+		AssetPrefix:      "aws-ebs-csi-driver",
+		AssetShortPrefix: "ebs",
+		DriverName:       "ebs.csi.aws.com",
+
+		ControllerConfig: &generator.ControlPlaneConfig{
+			DeploymentTemplateAssetName: "drivers/aws-ebs/patches/controller.yaml",
+			LivenessProbePort:           10301,
+			MetricsPorts: []generator.MetricsPort{
+				{
+					LocalPort:           common.AWSEBSLoopbackMetricsPortStart,
+					InjectKubeRBACProxy: true,
+					ExposedPort:         common.AWSEBSExposedMetricsPortStart,
+					Name:                "driver-m",
+				},
+			},
+			SidecarLocalMetricsPortStart:   common.AWSEBSLoopbackMetricsPortStart + 1,
+			SidecarExposedMetricsPortStart: common.AWSEBSExposedMetricsPortStart + 1,
+			Sidecars: []generator.SidecarConfig{
+				common.DefaultProvisionerWithSnapshots.WithExtraArguments(
+					"--default-fstype=ext4",
+					"--feature-gates=Topology=true",
+					"--extra-create-metadata=true",
+					"--timeout=60s",
```

Comment:
Let's say I want to set 120s for hypershift (for whatever reason) and 60s for standalone; with this structure, where's the best place to put that info?

Reply:
with the current structure it's not possible to have different args in standalone and in hypershift. I'd say we will add it when we need it?